### ISSUE TITLE
retinaface : INVALID ARGUMENT : can not find binding of given name 588,587.

### ISSUE THREAD

**Issue by chunniunai220ml** [Syencil/tensorRT#14]
ONNX IR version:  0.0.6
Opset version:    11
Producer name:    pytorch
Producer version: 1.5
Domain:           
Model version:    0

While parsing node number 108 [Resize]:
ERROR: ModelImporter.cpp:124 In function parseGraph:
[5] Assertion failed: ctx->tensors().count(inputName)
[08/12/2020-10:45:42] [E] Parsing File Failed
[08/12/2020-10:45:42] [E] Init Session Failed!
Segmentation fault (core dumped)

I have check the print export_onnx_model. no %108.

if I choose Opset version=12, the print of export_onnx_model.:%108:Long()
but,the same error counted

**Comment by chunniunai220ml** [comment]
I have solve this problem by onnx-simplifier
BUT, [E] [TRT] INVALID ARGUMENT : can  not find binding of given name 588,587.

conver2onnx:
  %581 : Long() = onnx::Gather[axis=0](%579, %580) # /mnt/MountVolume1/xielu/code_hub/codes/gitsfile/Pytorch_Retinaface-master/models/retinaface.py:46:0
  %584 : Tensor = onnx::Unsqueeze[axes=[0]](%581)
  %587 : Tensor = onnx::Concat[axis=0](%584, %607, %608)
  %588 : Float(1:8000, 800:10, 10:1) = onnx::Reshape(%578, %587) # /mnt/MountVolume1/xielu/code_hub/codes/gitsfile/Pytorch_Retinaface-master/models/retinaface.py:46:0
  %589 : Float(1:168000, 16800:10, 10:1) = onnx::Concat[axis=1](%564, %576, %588) # /mnt/MountVolume1/xielu/code_hub/codes/gitsfile/Pytorch_Retinaface-master/models/retinaface.py:121:0
  %590 : Float(1:33600, 16800:2, 2:1) = onnx::Softmax[axis=2](%552) # /usr/local/lib/python3.7/site-packages/torch/nn/functional.py:1498:0
  return (%output0, %590, %589)

python3 -m onnxsim FaceDetector11.onnx FaceDetector.onnx
Simplifying...
Checking 0/3...
Checking 1/3...
Checking 2/3...
Ok!

I have changed the outputTensorName{output0, 588, 587} in retinaface_mian.cpp to {output0, 590, 589}, also failed!

can u  tell me which version you choosed on torch,torchversion?
and i  compiled on tesla K80

**Comment by Syencil** [comment]
u should input the correct onnx outputs' name.

**Comment by chunniunai220ml** [comment]
I export the onnx.model by  your given link(pytorch_retinaface) with your tips(verbose=Ture,opset_version=11)

output_onnx = 'FaceDetector.onnx'
    print("==> Exporting model to ONNX format at '{}'".format(output_onnx))
    input_names = ["input0"]
    output_names = ["output0"]
    inputs = torch.randn(1, 3, args.long_side, args.long_side).to(device)

    torch_out = torch.onnx._export(net, inputs, output_onnx, export_params=True, verbose=Ture,
                                   input_names=input_names, output_names=output_names,opset_version=11) 

the print info :

%587 : Tensor = onnx::Concat[axis=0](%584, %607, %608)
%588 : Float(1:8000, 800:10, 10:1) = onnx::Reshape(%578, %587) # /mnt/MountVolume1/xielu/code_hub/codes/gitsfile/Pytorch_Retinaface-master/models/retinaface.py:46:0
%589 : Float(1:168000, 16800:10, 10:1) = onnx::Concat[axis=1](%564, %576, %588) # /mnt/MountVolume1/xielu/code_hub/codes/gitsfile/Pytorch_Retinaface-master/models/retinaface.py:121:0
%590 : Float(1:33600, 16800:2, 2:1) = onnx::Softmaxaxis=2 # /usr/local/lib/python3.7/site-packages/torch/nn/functional.py:1498:0
return (%output0, %590, %589)

I think the outputName should{"output0”, “590, ”589"}，but also failed,
 [E] [TRT] INVALID ARGUMENT : can not find binding of given name 590,589.

how can i find the right outputTensorName? used onnx.check and onnx.print?
how do you find the right outputTensorName?

**Comment by Syencil** [comment]
it is exactly (%output0, %590, %589).

**Comment by Syencil** [comment]
remove the serialized file and load the model from onnxfile

**Comment by chunniunai220ml** [comment]
> it is exactly (%output0, %590, %589).

I have set OutputTensorName ={"output0”, “590, ”589"} in retinaface_main.cpp, also error,INVALID ARGUMENT !!!

and I  use this code in pytorch documentation  to check FaceDetector.onnx

import onnx

# Load the ONNX model
model = onnx.load("FaceDetector.onnx")
print(model)
# Check that the IR is well formed
onnx.checker.check_model(model)

# Print a human readable representation of the graph
onnx.helper.printable_graph(model.graph)

the proto imshow  the outputtensorname is :  output0,590,589
input {
  name: "input0"
  type {
    tensor_type {
      elem_type: 1
      shape {
        dim {
          dim_value: 1
        }
        dim {
          dim_value: 3
        }
        dim {
          dim_value: 640
        }
        dim {
          dim_value: 640
        }
      }
    }
  }
}
output {
  name: "output0"
  type {
    tensor_type {
      elem_type: 1
      shape {
        dim {
          dim_value: 1
        }
        dim {
          dim_value: 16800
        }
        dim {
          dim_value: 4
        }
      }
    }
  }
}
output {
  name: "590"
  type {
    tensor_type {
      elem_type: 1
      shape {
        dim {
          dim_value: 1
        }
        dim {
          dim_value: 16800
        }
        dim {
          dim_value: 2
        }
      }
    }
  }
}
output {
  name: "589"
  type {
    tensor_type {
      elem_type: 1
      shape {
        dim {
          dim_value: 1
        }
        dim {
          dim_value: 16800
        }
        dim {
          dim_value: 10
        }
      }
    }
  }
}

**Comment by chunniunai220ml** [comment]
> remove the serialized file and load the model from onnxfile

what is your means? I just have modelxxx.onnx,  load the model from onnxfile with python or C++?

**Comment by Syencil** [comment]
Check the details carefully.

**Comment by chunniunai220ml** [comment]
> Check the details carefully.

can u tell me which version you have uesed for torch,onnx,tensorRT7.0?  there may be some incompatible problems so that the parser can not find the nodes

**Comment by Syencil** [comment]
和这个没有关系 你去看看buffer.h这个文件 从engine的索引获取输入输出名称 查一下你engine文件就行了
